ext/http/00_serve.js

```diff
@@ -331,11 +331,28 @@ function fastSyncResponseOrStream(req, respBody) {
 }
 
 async function asyncResponse(responseBodies, req, status, stream) {
+  const reader = stream.getReader();
+  // Optimize for streams that are done in zero or one packets. We will not
+  // have to allocate a resource in this case.
+  const { value: value1, done: done1 } = await reader.read();
+  if (done1) {
+    core.ops.op_set_promise_complete(req, status);
+    return;
+  }
+  const { value: value2, done: done2 } = await reader.read();
+  if (done2) {
+    core.ops.op_set_response_body_bytes(req, value1);
+    core.ops.op_set_promise_complete(req, status);
+    return;
+  }
```

Comment:
If these reads throw, bad things happen. This should happen inside of the try-catch-finally.

Reply:
Agreed, otherwise LGTM

Reply:
I reworked this pretty massively and added throwing stream tests.